sklearn/datasets/tests/test_base.py

```diff
@@ -180,6 +181,16 @@ def test_load_iris():
     assert_equal(res.target_names.size, 3)
     assert_true(res.DESCR)
 
+    # test return_X_y option
+    X_y_tuple = load_iris(return_X_y=True)
+    bunch = load_iris()
+    assert_true(isinstance(X_y_tuple, tuple))
+    assert_equal(X_y_tuple[0].shape, bunch.data.shape)
```

Comment:
I think this is not what @jnothman meant.
You can actually check the arrays:
```python
assert_array_equal(X_y_tuple[0], bunch.data)
assert_array_equal(X_y_tuple[1], bunch.target)
```
which checks at the same time the shape, and the data.